app/controllers/bots_controller.rb

```diff
@@ -64,6 +64,83 @@ def destroy
       format.json { head :no_content }
     end
   end
+  
+  
+  # POST /bots/1/collaborators
+  # POST /bots/1/collaborators.json
+  def add_collaborator
+      unless Bot.exists?(id: params[:bot])
+        respond_to do |format|
+          format.html { redirect_to index_path, notice: 'That bot ID does not exist.' }
+          format.json { render json: 'That bot ID does not exist.', status: :unprocessable_entity }
+        end
+        return
+      end
+      bot = Bot.find(params[:bot])
```

Comment:
You don't actually need the `unless` block here: calling `find` will result in a 404 response if the ID doesn't exist, which is pretty much ideal. Same thing on lines 88, 117, and 126.